forem/forem
# Fix inconsistent published date of Article and Comment

## What type of PR is this? (check all applicable)


## Description
This PR fixes the bug that the published date of comments may appear to be before the article's.
The bug occurs because UTC time is shown for comments while local time is shown for the article's posted date.
Therefore, this PR changes the published date of comments to be shown in local time.
 
## Related Tickets & Documents


- Closes #16744

## QA Instructions, Screenshots, Recordings
To avoid false positive results, the date of your local time must be different than the date of UTC time. For example, if your local time zone is Eastern Standard Time, you can run this test from 7pm - midnight (from 8 pm if Daylight Savings Time). 
You can either wait for an opportune time or change your system's time zone temporarily.

1. Create a new post.
2. Create a new comment under the new post.
3. Verify that the article's posted date is the same as the comment's.
4. Refresh the page, and verify the dates again. You should see the dates match. See example below:  ![fix comment date 1](https://user-images.githubusercontent.com/88938117/188131485-eda50b92-c51d-4714-b9c4-e017f311f7fe.png)
5. Edit the comment and verify the "edited on" date is consistent.
6. Refresh the page and verify it again.

The second part of this QA verifies that the format of the comment date is not affected by this PR, especially when the comment is from previous years.

1. First, create at least a post and a comment that are from the year before. There may be better ways of doing it, but I seeded the database with the help of Timecop. 
2. Navigate to an article posted last year.
3. Verify that the format of comment dates has not changed. See example below: 
![fix comment date 2](https://user-images.githubusercontent.com/88938117/188139181-36879f93-1341-440f-b4f6-4eb37d5fc78f.png)

### UI accessibility concerns?

UI accessibility is not affected by this PR.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
